src/test/java/io/jenkins/plugins/analysis/core/history/PreviousRunReferenceTest.java

```diff
@@ -458,25 +451,34 @@ void allTypesOfAction_withAnyResult_shouldReturnThree() {
         PreviousRunReference previousRun = new PreviousRunReference(baseline, selector,
                 false);
 
-        Optional<ResultAction> resultActionFAILURE = previousRun.getReferenceAction();
-        Optional<ResultAction> resultActionUNSTABLE = previousRun.getReferenceAction();
-        Optional<ResultAction> resultActionSUCCESS = previousRun.getReferenceAction();
-        Optional<ResultAction> resultActionNOT_BUILT = previousRun.getReferenceAction();
-        Optional<ResultAction> resultActionABORTED = previousRun.getReferenceAction();
-
-        verify(selector, times(8)).get(previousRunMock);
-        assertThat(resultActionFAILURE).isEqualTo(Optional.of(action));
-        assertThat(resultActionUNSTABLE).isEqualTo(Optional.of(action));
-        assertThat(resultActionSUCCESS).isEqualTo(Optional.of(action));
-        assertThat(resultActionNOT_BUILT).isEmpty();
-        assertThat(resultActionABORTED).isEmpty();
+        when(previousRunMock.getResult()).thenReturn(Result.FAILURE);
+        when(analysis.getOverallResult()).thenReturn(Result.FAILURE);
+        assertThat(previousRun.getReferenceAction()).isEqualTo(Optional.of(action));
+
+        when(previousRunMock.getResult()).thenReturn(Result.UNSTABLE);
+        when(analysis.getOverallResult()).thenReturn(Result.UNSTABLE);
+        assertThat(previousRun.getReferenceAction()).isEqualTo(Optional.of(action));
+
+        when(previousRunMock.getResult()).thenReturn(Result.SUCCESS);
+        when(analysis.getOverallResult()).thenReturn(Result.SUCCESS);
+        assertThat(previousRun.getReferenceAction()).isEqualTo(Optional.of(action));
+
+        when(previousRunMock.getResult()).thenReturn(Result.NOT_BUILT);
+        when(analysis.getOverallResult()).thenReturn(Result.NOT_BUILT);
+        assertThat(previousRun.getReferenceAction()).isEqualTo(Optional.of(action));
+
+        when(previousRunMock.getResult()).thenReturn(Result.ABORTED);
+        when(analysis.getOverallResult()).thenReturn(Result.ABORTED);
+        assertThat(previousRun.getReferenceAction()).isEqualTo(Optional.of(action));
+
```

Comment:
Why don't you test some combinations of different result and overallresult (maybe in another method)
They don't have to be the same, and in some combinations they will do different stuff.
i.e. Result == Failure + OverallResult == Success and Result == Failure + OverallResult == Failure are behaving differently in some cases (see: BuildHistory.class)